crates/config/src/config.rs

```diff
@@ -48,9 +48,13 @@ impl Config {
             return Err(std::io::Error::new(
                 std::io::ErrorKind::InvalidInput,
                 format!("reth config file extension must be '{EXTENSION}'"),
-            ))
+            ));
         }
-        confy::store_path(path, self).map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e))
+
+        let config = toml::to_string(self)
+            .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e.to_string()))?;
+        std::fs::write(path, config)
+            .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e.to_string()))
```

Comment:
can we map it to a proper write error here?

Reply:
Just fixed, I think at the end we don't even need to map as `std::fs::write` is giving us the right error format imo